File: spat-server/src/main/java/com/pcalouche/spat/api/team/service/TeamService.java
package com.pcalouche.spat.api.team.service;

import com.pcalouche.spat.api.model.Team;

import java.util.List;

public interface TeamService {
    List<Team> getTeams();

    Team saveTeam(Team team);

    Boolean deleteTeam(Long id);
}

File: angular-ui/src/app/rest-services/api/team/team.service.ts
import {HttpClient}        from '@angular/common/http';
import {Injectable}        from '@angular/core';
import {Team}              from '@rest-services/api/model/team.model.a';
import {RestServiceHelper} from '@rest-services/api/rest-service-helper';
import {Observable}        from 'rxjs';

@Injectable()
export class TeamService {
  private readonly teamRoot = RestServiceHelper.apiRoot + '/teams';

  constructor(private http: HttpClient) {
  }

  teams(): Observable<Team[]> {
    return this.http.get<Team[]>(this.teamRoot);
  }

  saveTeam(team: Team): Observable<Team> {
    return this.http.post<Team>(this.teamRoot, team);
  }

  deleteTeam(team: Team): Observable<boolean> {
    return this.http.delete<boolean>(this.teamRoot + '/' + team.id);
  }
}
